src/lib/utils/track.ts

```diff
@@ -201,3 +218,104 @@ export interface Track<P = any, S = null, T extends Schema.Global = Schema.Entit
  *      ```
  */
 export const track: Track = _track
+
+/**
+ * A typed page view decorator for the top level component for your screen. This is the
+ * function you use at the root of your component tree.
+ *
+ * For the majority of Emission code, this should only be used inside the AppRegistry,
+ * however if you have other components which are going to be presented using a navigation
+ * controller then you'll need to use this.
+ *
+ * The main implementation difference between this and `track` is that this hooks the callbacks
+ * to our native `Events.postEvent` function.
+ *
+ * As an object:
+ *
+ * @example
+ *
+ *      ```ts
+ *      import { screenTrack, Schema } from "lib/utils/track"
+ *
+ *       @screenTrack({
+ *        context_screen: Schema.PageNames.ConsignmentsWelcome,
+ *        context_screen_owner_slug: null,
+ *        context_screen_owner_type: Schema.OwnerEntityTypes.Consignment,
+ *       })
+ *
+ *       export default class Welcome extends React.Component<Props, null> {
+ *         [...]
+ *
+ * * As an function taking account of incoming props:
+ *
+ * @example
+ *
+ *      ```ts
+ *      import { screenTrack, Schema } from "lib/utils/track"
+ *
+ *      interface Props extends ViewProperties {
+ *        [..]
+ *      }
+ *
+ *      @screenTrack<Props>(props => ({
+ *        context_screen: Schema.PageNames.ConsignmentsSubmission,
+ *        context_screen_owner_slug: props.submissionID,
+ *        context_screen_owner_type: Schema.OwnerEntityTypes.Consignment,
+ *      }))
+ *      export default class Welcome extends React.Component<Props, null> {
+ *
+ *        [...]
```

Comment:
The code block formatting of this example seems broken, as it’s missing trailing backticks. And maybe replace `[...]` with `// ...` so that it’s still valid code?

Reply:
Same goes for the example above btw.